scripts_v2/Dockerfile.build

```diff
@@ -1,13 +1,23 @@
 # syntax = docker/dockerfile:experimental
 
-# this fixes the version of the golang image to avoid breaking changes in glibc
-# sha1 comes from https://github.com/docker-library/repo-info/blob/b939681d85e36bae10eabe53b913f1d05e972f1f/repos/golang/remote/1.19.md
-FROM golang@sha256:2dde7ccba42da98d45e0f38f23f1c0d1474a779e4089faf665f9af0dd3f844db as builder
+FROM public.ecr.aws/lambda/provided:al2 as builder
 ARG EXTENSION_VERSION
 ARG AGENT_VERSION
 ARG CMD_PATH
 ARG BUILD_TAGS
 
+# install go 1.19
+RUN yum install -y wget tar gzip gcc
+RUN arch="$(uname -m)"; \
+    if [ "${arch}" = 'aarch64' ]; then \
+        arch='arm64'; \
+    fi; \
+    if [ "${arch}" = 'x86_64' ]; then \
+        arch='amd64'; \
+    fi; \
```

Comment:
alternatively, you could pass arch in as a build arg.

Reply:
yeah definitely, I'll leave this like that for now to be sure not to break backward compatibility